src/js/decafmud.js

```diff
@@ -1532,6 +1565,16 @@ DecafMUD.prototype.socketClosed = function() {
 /** Called by the socket when data arrives. */
 DecafMUD.prototype.socketData = function(data) {
 	// Push the text onto the inbuf.
+	if (this.compressingv2) { // Decompress it first!
+		try {
+			data = this.decompressStream.decompress(data);
+		} catch (e) {
+			console.error('MCCP2 compression disabled because ' + e);
```

Comment:
Should we instead use this.decaf.debugString() or this.decaf.error() ? The later will show an error to the end-user which might not be very useful.

Reply:
https://github.com/Tiger-Tom/DecafMUD/commit/25922361baa093d3d776952f18616eba68a0ad20